internal/vault/vault.go

```diff
@@ -403,13 +403,24 @@ func (v *Vault) requestTokenWithKubernetesAuth(client Client, kubernetesAuth *v1
 		}
 		aud += v.issuer.GetName()
 
+		audiences := append(kubernetesAuth.TokenAudiences, aud)
+
```

Comment:
suggestion: Do you think it works if we rename "aud" to "defaultAudience" in this section, to make the difference between `aud` and `audiences` clearer?